src/scripts/main.js

```diff
@@ -1,5 +1,12 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const img = document.querySelector('.spider');
+
+  if (!e.target.closest('.wall')) {
+    return;
+  }
+
+  img.style.top = e.offsetY - img.clientHeight / 2 + 'px';
```

Comment:
The calculation for the spider's position does not account for the page's scroll position. Consider using `e.clientY` and `e.clientX` instead of `e.offsetY` and `e.offsetX` to ensure the position is calculated correctly relative to the viewport.